cmd/mps-control-daemon/mps/daemon.go

```diff
@@ -124,23 +128,38 @@ func (d *Daemon) Start() error {
 	}
 	defer statusFile.Close()
 
+	ctx, cancel := context.WithCancel(context.Background())
+	d.tailCancel = cancel
+
+	klog.InfoS("Starting log tailer", "resource", d.rm.Resource())
+	//nolint:gosec // G204: Subprocess launched with a potential tainted input or cmd arguments (gosec)
+	tail := exec.CommandContext(ctx, "tail", "-n", "+1", "-f", filepath.Join(logDir, "control.log"))
+	tail.Stdout = os.Stdout
+	tail.Stderr = os.Stderr
+
+	if err := tail.Start(); err != nil {
+		klog.ErrorS(err, "Could not start tail command on control.log; ignoring logs")
+	}
+
 	return nil
 }
 
 // Stop ensures that the MPS daemon is quit.
 func (d *Daemon) Stop() error {
-	output, err := d.EchoPipeToControl("quit")
+	_, err := d.EchoPipeToControl("quit")
 	if err != nil {
 		return fmt.Errorf("error sending quit message: %w", err)
 	}
-	klog.InfoS("Shut down MPS", "output", output)
+	klog.InfoS("Stopped MPS control daemon", "resource", d.rm.Resource())
+
+	klog.InfoS("Stopping log tailer", "resource", d.rm.Resource())
+	d.tailCancel()
```

Comment:
Thinking about this a bit more (after looking at it), we should probably still track `d.tail` and call `d.tail.Wait()` after the cancel(). It is guaranteed to return and we can check / log its error (if there is one).

Reply:
Maybe call it `d.tailCmd` though, so as to by symmetric with `d.tailCancel`